launcher/launcher.go

```diff
@@ -687,46 +687,70 @@ func (launcher *Launcher) getLayersForService(digests []string) ([]imagemanager.
 	return layers, nil
 }
 
-func getStorageRequestRatio(
-	serviceRatios *aostypes.ResourceRatiosInfo, nodeRatios *aostypes.ResourceRatiosInfo,
-) float64 {
-	if serviceRatios != nil && serviceRatios.Storage != nil {
-		return float64(*serviceRatios.Storage) / 100
+func getReqDiskSize(serviceConfig aostypes.ServiceConfig, nodeRatios *aostypes.ResourceRatiosInfo,
+) (stateSize, storageSize uint64) {
+	stateQuota := serviceConfig.Quotas.StateLimit
+	storageQuota := serviceConfig.Quotas.StorageLimit
+
+	if serviceConfig.RequestedResources != nil && serviceConfig.RequestedResources.Storage != nil {
+		stateSize = clampResource(*serviceConfig.RequestedResources.Storage, stateQuota)
+		storageSize = clampResource(*serviceConfig.RequestedResources.Storage, storageQuota)
+	} else {
+		stateSize = getReqStorageFromNodeConf(stateQuota, nodeRatios)
+		storageSize = getReqStorageFromNodeConf(storageQuota, nodeRatios)
 	}
 
-	if nodeRatios != nil && nodeRatios.Storage != nil {
-		return float64(*nodeRatios.Storage) / 100
+	return
```

Comment:
Is this return required?

Reply:
yes, "missing return " error have otherwise

Reply:
fixed